opm/input/eclipse/Schedule/UDQ/UDQAssign.hpp

```diff
@@ -17,89 +17,127 @@
   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-
 #ifndef UDQASSIGN_HPP_
 #define UDQASSIGN_HPP_
 
+#include <opm/input/eclipse/Schedule/UDQ/UDQEnums.hpp>
+#include <opm/input/eclipse/Schedule/UDQ/UDQSet.hpp>
+
+#include <cstddef>
 #include <string>
 #include <unordered_set>
+#include <utility>
 #include <vector>
 
-#include <opm/input/eclipse/Schedule/UDQ/UDQSet.hpp>
-#include <opm/input/eclipse/Schedule/UDQ/UDQEnums.hpp>
-
 namespace Opm {
 
-
-class UDQAssign{
+class UDQAssign
+{
 public:
-
-    /*
-      If the same keyword is assigned several times the different assignment
-      records are assembled in one UDQAssign instance. This is an attempt to
-      support restart in a situation where a full UDQ ASSIGN statement can be
-      swapped with a UDQ DEFINE statement.
-    */
-    struct AssignRecord {
-        std::vector<std::string> input_selector;
-        std::unordered_set<std::string> rst_selector;
-        double value;
-        std::size_t report_step;
+    // If the same keyword is assigned several times the different
+    // assignment records are assembled in one UDQAssign instance.  This is
+    // an attempt to support restart in a situation where a full UDQ ASSIGN
+    // statement can be swapped with a UDQ DEFINE statement.
+    struct AssignRecord
+    {
+        std::vector<std::string> input_selector{};
+        std::unordered_set<std::string> rst_selector{};
+        std::vector<UDQSet::EnumeratedWellItems> numbered_selector{};
```

Comment:
I know that you did not touch this besides adding stuff, but this might a good time to share knowledge about the different selectors (e.g. are they exclusive as in only one may have values)?

Reply:
> are they exclusive as in only one may have values

Yes they are.  I suppose we might wrap them in a `variant<>` to  highlight that fact.  I'd prefer to do that in follow-up work though.

Reply:
that sounds nice.